std::vector<unsigned char> profile(profile_size) is not compatible with typedef char JOCTET

**Describe the bug**
`std::vector<unsigned char> profile(profile_size)` is not compatible with `typedef char JOCTET`
```
oiio/src/jpeg.imageio/jpegoutput.cpp:323:60: error: invalid conversion from ‘__gnu_cxx::__alloc_traits<std::allocator<unsigned char>, unsigned char>::value_type*’ {aka ‘unsigned char*’} to ‘const JOCTET*’ {aka ‘const char*’} [-fpermissive]
  323 |                 jpeg_write_marker(&m_cinfo, JPEG_APP0 + 2, &profile[0],
```
Both the Visual Code Studio and the gcc compiler complains about that.

**To Reproduce**
Open on Visual Code Studio or try to build with cmake.

**Expected behavior**
Correct usage of the type for the profile declaration.

**Evidence**
```
oiio/src/jpeg.imageio/jpegoutput.cpp:323:60: error: invalid conversion from ‘__gnu_cxx::__alloc_traits<std::allocator<unsigned char>, unsigned char>::value_type*’ {aka ‘unsigned char*’} to ‘const JOCTET*’ {aka ‘const char*’} [-fpermissive]
  323 |                 jpeg_write_marker(&m_cinfo, JPEG_APP0 + 2, &profile[0],
```

**Platform information:**
 - OIIO branch/version: recent from git (v2_5_0)
 - OS: Ubuntu 20.04
 - C++ compiler: gcc-9.4.0
 - Any non-default build flags when you build OIIO: none

### Note
It now compiles changing `std::vector<unsigned char> profile(profile_size)` by `std::vector<char> profile(profile_size)` but now running `"imageinout_test"` I receive: 
`JPEG parameter struct mismatch: library thinks size is 584, caller expects 640`

--- a/src/jpeg.imageio/jpegoutput.cpp
+++ b/src/jpeg.imageio/jpegoutput.cpp
@@ -305,7 +305,7 @@ JpgOutput::open(const std::string& name, const ImageSpec& newspec,
                 num_markers++;
             int curr_marker     = 1; /* per spec, count starts at 1*/
             size_t profile_size = MAX_DATA_BYTES_IN_MARKER + ICC_HEADER_SIZE;
-            std::vector<unsigned char> profile(profile_size);
+            std::vector<JOCTET> profile(profile_size);
             while (icc_profile_length > 0) {
                 // length of profile to put in this marker
                 unsigned int length
@@ -316,11 +316,11 @@ JpgOutput::open(const std::string& name, const ImageSpec& newspec,
                 strcpy((char*)profile.data(), "ICC_PROFILE");  // NOSONAR
                 profile[11] = 0;
                 profile[12] = curr_marker;
-                profile[13] = (unsigned char)num_markers;
+                profile[13] = (JOCTET)num_markers;
                 memcpy(profile.data() + ICC_HEADER_SIZE,
                        icc_profile + length * (curr_marker - 1),
                        length);  //NOSONAR
-                jpeg_write_marker(&m_cinfo, JPEG_APP0 + 2, &profile[0],
+                jpeg_write_marker(&m_cinfo, JPEG_APP0 + 2, profile.data(),
                                   ICC_HEADER_SIZE + length);
                 curr_marker++;
             }

